File: xindice/ringo/update/xindice_update.js
#! /usr/bin/ringo
// ------------------------------------------------------------------
//
//	xindice_update.js
//
//						May/18/2011
//
// ------------------------------------------------------------------
var text_manipulate = require ('/var/www/data_base/common/ringo_common/text_manipulate.js');
var xml_manipulate = require ('/var/www/data_base/common/ringo_common/xml_manipulate.js');
// ------------------------------------------------------------------
print	("*** 開始 ***");
var args = require('system').args;
var id_in = args[1];
var population_in = args[2];
print	(id_in + "\t" + population_in);

var client= require ("ringo/httpclient");

var url='http://host_dbase:8888/xindice/db/cities/cities';

print	("*** ppp ***");
var data=client.get (url);
print	("*** qqq ***");

print (data.status);
// print (data.content);

var dict_aa = xml_manipulate.xml_to_dict_proc (data.content);

text_manipulate.dict_display_proc (dict_aa);

dict_aa = text_manipulate.dict_update_proc (dict_aa,id_in,population_in);

var out_str = xml_manipulate.dict_to_xml_proc (dict_aa);

// print (out_str);

client.put (url,out_str);

print	("*** 終了 ***");
// ------------------------------------------------------------------

File: xindice/cplus/update/xindice_update.cpp
// --------------------------------------------------------------------
/*
	xindice_update.cpp

				Feb/28/2012


*/
// --------------------------------------------------------------------
#include	<string>
#include	<cstring>
#include	<iostream>
#include	<map>
#include	<curl/curl.h>

#include <stdio.h>
#include <stdlib.h>

using namespace std;
typedef map<string,string> Unit;

extern string url_get_proc (char url[]);
extern void curl_put_proc (char *url_target,string str_in);
extern	void dict_display_proc (map <string,Unit> dict_aa);
extern	map <string,Unit> xml_to_dict_proc (string str_xml);
extern map <string,Unit> dict_update_proc
	(map <string,Unit> dict_aa,char id_in[],int population_in);
extern string dict_to_xml_proc (map <string,Unit> dict_aa);
// --------------------------------------------------------------------
int main (int argc,char *argv[])
{
	cerr << "*** 開始 ***\n";

	char key_in[10];
	strcpy (key_in,argv[1]);
	int population_in = atoi (argv[2]);

	cout << key_in << '\t';
	cout << population_in << endl;

	char url_target[] = "http://host_dbase:8888/xindice/db/cities/cities";

	string str_xml = url_get_proc (url_target);

	map <string,Unit> dict_aa =  xml_to_dict_proc (str_xml);

	dict_aa = dict_update_proc (dict_aa,key_in,population_in);

	dict_display_proc (dict_aa);

	str_xml = dict_to_xml_proc (dict_aa);

	cerr << "*** ccc ***\n";

	curl_put_proc (url_target,str_xml);

	cerr << "*** 終了 ***\n";

	return 0;
}

// --------------------------------------------------------------------
